e2e/e2e_test.go

```diff
@@ -502,6 +502,87 @@ func TestE2E_ConfigStreamlining_Deprecations(t *testing.T) {
 	validateServices(t, true, []string{"api"}, resourcesPath)
 }
 
+// TestE2E_TriggerTaskWhenActive tests making changes to monitored dependencies
+// while a task is running. It verifies:
+//
+// 1. a change to the same task will trigger the task after the run completes
+// 2. a change to a different task runs immediately
+// 3. a change after the task completes will still trigger the task to run
+//
+// https://github.com/hashicorp/consul-terraform-sync/issues/732
+func TestE2E_TriggerTaskWhenActive(t *testing.T) {
+	// Start CTS with multiple tasks
+	activeTaskName := "active_task"
+	activeTaskConfig := fmt.Sprintf(`
+	task {
+		name = "%s"
+		providers = ["local"]
+		module = "./test_modules/delayed_module"
+		condition "services" {
+			names = ["api"]
+		}
+	}
+	`, activeTaskName)
+
+	inactiveTaskName := "inactive_task"
+	inactiveTaskConfig := fmt.Sprintf(`
+	task {
+		name = "%s"
+		providers = ["local"]
+		module = "./test_modules/local_instances_file"
+		condition "services" {
+			names = ["web"]
+		}
+	}
+	`, inactiveTaskName)
+
+	srv := testutils.NewTestConsulServer(t, testutils.TestConsulServerConfig{
+		HTTPSRelPath: "../testutils",
+	})
+	defer srv.Stop()
+	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "trigger_when_active")
+	cts := ctsSetup(t, srv, tempDir, activeTaskConfig, inactiveTaskConfig)
+
+	// Trigger the task with the long-running module so that it becomes active
+	firstRegister := time.Now()
+	testutils.RegisterConsulService(t, srv,
+		testutil.TestService{Name: "api", ID: "api-1"}, defaultWaitForRegistration)
+
+	// Trigger the task again while still running
+	time.Sleep(1 * time.Second) // waiting for task to start, module sleeps for 5 seconds
+	activeRegister := time.Now()
+	testutils.RegisterConsulService(t, srv,
+		testutil.TestService{Name: "api", ID: "api-2"}, defaultWaitForRegistration)
+
+	// Trigger the other task while the other task is running
+	inactiveRegister := time.Now()
+	testutils.RegisterConsulService(t, srv,
+		testutil.TestService{Name: "web", ID: "web-1"}, defaultWaitForRegistration)
+
+	// Other task should complete without waiting for first task to complete
+	api.WaitForEvent(t, cts, inactiveTaskName, inactiveRegister, defaultWaitForEvent)
+	count := eventCount(t, inactiveTaskName, cts.Port())
+	assert.Equal(t, 2, count, "unexpected number of events")
```

Comment:
Oh, I just realized I didn't add any checks to make sure the module output reflected the expected services! Events are probably sufficient, but I'll add it just in case!